### ISSUE TITLE
Unable to process UDP responses

### ISSUE THREAD

**Issue by ptoews** [EmperorCookie/accapi#2]
Hi,

first of all this wrapper looks great and well written, and I would like to use it for a project.
However I could not get it running so far. After fixing the game crashing as in #1, I'm now stuck with the following exception being thrown during the registration process:
```
  File "D:\Development\accapi\src\accapi\client.py", line 283, in _run
    messageType = self._receive("B")
  File "D:\Development\accapi\src\accapi\client.py", line 137, in _receive
    val, = struct.unpack(f"!{f}", self._socket.recv(struct.calcsize(f)))
OSError: [WinError 10040] Eine Nachricht, die über einen Datagrammsocket gesendet wurde, war für den internen Nachrichtenpuffer oder ein anderes Netzwerklimit zu groß, oder der Puffer für den Datagrammempfang war für das Datagramm zu klein
```
where the description roughly translates to a hint that a message buffer was too small in order to receive a datagram package.

I don't know if I'm missing something, but from my understanding a UDP datagram has to be received with a `recv()` call with a buffer size parameter that is at least as large as the incoming datagram, otherwise it is discarded which would explain above error message.

So the error here seems to be that the incoming data is received in steps, starting with only the first byte to identify the message type. The solution would be to first receive the full datagram and then do the processing in steps. I'm just wondering how this has worked previously, if it has, or if I'm missing something?

**Comment by EmperorCookie** [comment]
Thanks for the kind words! I will investigate and fix this ASAP.

**Comment by ptoews** [comment]
I've already rewritten parts of the code now in order to process the incoming messages after receiving the full datagram, and was then able to read the data from the game. During that I had to fix a few other minor bugs as well, so I might push a PR later if you're interested.

**Comment by EmperorCookie** [comment]
I had a look at your fork and it looks like the changes you're talking about were not committed/pushed. As for the problem itself, I suspect the byte order and data type fix in the other PR would be the culprit here as well.

**Comment by EmperorCookie** [comment]
This issue should be completely fixed once #4 is merged.
